helpers/image/image_test.go

```diff
@@ -17,209 +17,197 @@ package image
+			in:   "test.com:8080/namespace/name:v1@sha256:1234567890",
+			expect: expectations{
+				Hostname:        "test.com",
+				HasIBMRepo:      false,
+				Port:            "8080",
+				Tag:             "v1",
+				Digest:          "1234567890",
+				NameWithTag:     "test.com:8080/namespace/name:v1",
+				NameWithoutTag:  "test.com:8080/namespace/name",
+				String:          "test.com:8080/namespace/name:v1@sha256:1234567890",
+				RegistryURL:     "https://test.com:8080",
+				ContentTrustErr: true,
+			},
+		},
+		{
+			name: "parses an image from Docker Hub with a tag and a digest",
+			in:   "namespace/name:v1@sha256:1234567890",
+			expect: expectations{
+				Hostname:        "docker.io",
+				HasIBMRepo:      false,
+				Port:            "",
+				Tag:             "v1",
+				Digest:          "1234567890",
+				NameWithTag:     "namespace/name:v1",
+				NameWithoutTag:  "namespace/name",
+				String:          "namespace/name:v1@sha256:1234567890",
+				RegistryURL:     "https://docker.io",
+				ContentTrustErr: false,
+				ContentTrustURL: "https://notary.docker.io",
+			},
+		},
+		{
+			name: "parses a Docker Hub public image with a tag and a digest",
+			in:   "ubuntu:v1@sha256:1234567890",
+			expect: expectations{
+				Hostname:        "docker.io",
+				HasIBMRepo:      false,
+				Port:            "",
+				Tag:             "v1",
+				Digest:          "1234567890",
+				NameWithTag:     "ubuntu:v1",
+				NameWithoutTag:  "ubuntu",
+				String:          "ubuntu:v1@sha256:1234567890",
+				RegistryURL:     "https://docker.io",
+				ContentTrustErr: false,
+				ContentTrustURL: "https://notary.docker.io",
+			},
+		},
+		{
+			name: "parses an IBM image",
+			in:   "registry.ng.bluemix.net/namespace/name",
+			expect: expectations{
+				Hostname:        "registry.ng.bluemix.net",
+				HasIBMRepo:      true,
+				Port:            "",
+				Tag:             "latest",
+				Digest:          "",
+				NameWithTag:     "registry.ng.bluemix.net/namespace/name:latest",
+				NameWithoutTag:  "registry.ng.bluemix.net/namespace/name",
+				String:          "registry.ng.bluemix.net/namespace/name",
+				RegistryURL:     "https://registry.ng.bluemix.net",
+				ContentTrustErr: false,
+				ContentTrustURL: "https://registry.ng.bluemix.net:4443",
+			},
+		},
+		{
+			name: "parses a quay.io image",
+			in:   "quay.io/namespace/name",
+			expect: expectations{
+				Hostname:        "quay.io",
+				HasIBMRepo:      false,
+				Port:            "",
+				Tag:             "latest",
+				Digest:          "",
+				NameWithTag:     "quay.io/namespace/name:latest",
+				NameWithoutTag:  "quay.io/namespace/name",
+				String:          "quay.io/namespace/name",
+				RegistryURL:     "https://quay.io",
+				ContentTrustErr: false,
+				// TODO the previous ginkgo test was testing for port 443, but the code returns 4443. Need to verify which is correct.
+				// ContentTrustURL: "https://quay.io:443",
+				ContentTrustURL: "https://quay.io:4443",
```

Comment:
I'm not sure which URL is correct for Quay's notary server. https://quay.io:4443 gives connection refused, and https://quay.io:443 is the same as the registry URL.

The old test said port 443 (although the test never ran), and the code returns 4443.

@ecordell @ericavonb please could one of you confirm which port is the one that we should be using for notary information about images from Quay?